src/recursive_sorting/recursive_sorting.py

```diff
@@ -1,17 +1,34 @@
 # TO-DO: complete the helpe function below to merge 2 sorted arrays
-def merge( arrA, arrB ):
-    elements = len( arrA ) + len( arrB )
-    merged_arr = [0] * elements
+def merge(arrA, arrB):
+    # elements = len(arrA) + len(arrB)
+    # merged_arr = [0] * elements
+    merged_arr = []
+    left, right = 0, 0
     # TO-DO
-    
+    while left < len(arrA) and right < len(arrB):
+        if arrA[left] < arrB[right]:
+            merged_arr.append(arrA[left])
+            left += 1
+        else:
+            merged_arr.append(arrB[right])
+            right += 1
+    if left == len(arrA):
+        merged_arr.extend(arrB[right:])
+    else:
+        merged_arr.extend(arrA[left:])
+
     return merged_arr
 
 
 # TO-DO: implement the Merge Sort function below USING RECURSION
-def merge_sort( arr ):
+def merge_sort(arr):
     # TO-DO
-
-    return arr
+    if len(arr) <= 1:
```

Comment:
its funny that almost all these sorts is finding the middle and doing the work recursively, huh?